mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -900,6 +903,25 @@ private static DBObject stage(String key, Object value) {
 		return stage;
 	}
 
+	private static ClosableIterator<Tuple> doDistinct(
+			final MongoDBQueryDescriptor queryDescriptor, final DBCollection collection) {
+		List distinctFieldValues = collection.distinct( queryDescriptor.getFieldName(), queryDescriptor.getCriteria() );
```

Comment:
I see that the distinct method also has a ReadPreference option, could you check if it make sense to pass the argument returned by `tupleContext.getTupleTypeContext().getOptionsContext().getUnique( ReadPreferenceOption.class );`